### PR TITLE
[pkp-lib] #5640 Add support for single context site settings form

### PR DESCRIPTION
**Pull request by defstat**

### CODE REVIEW

File: templates/admin/settings.tpl [hunk 1]
```diff
@@ -12,54 +12,72 @@
...
 				{call_hook name="Template::Settings::admin::appearance"}
 			</tabs>
 		</tab>
+		{/if}
+		{if $componentAvailability['SitePlugins']}
 		<tab id="plugins" label="{translate key="common.plugins"}">
 			{capture assign=pluginGridUrl}{url router=$smarty.const.ROUTE_COMPONENT component="grid.admin.plugins.AdminPluginGridHandler" op="fetchGrid" escape=false}{/capture}
 			{load_url_in_div id="pluginGridContainer" url=$pluginGridUrl}
 		</tab>
+		{/if}
```
Comment (NateWr) [inline templates/admin/settings.tpl:80]: I wonder if we need this level of abstraction in the template. Wouldn't it be enough to have a single template boolean, `hasMultipleContexts`, and use that to hide the tabs we don't want?

With the component availability approach, it's possible to accidentally get yourself into a case where a parent tab is meant to be hidden but a child tab is meant to be shown.

But I'm ok leaving it as it is if you think it's better this way.

File: pages/admin/AdminHandler.inc.php [hunk 2]
```diff
@@ -118,10 +102,45 @@ function settings($args, $request) {
...
+			'SiteConfig', // Side tab, 'SiteSetup' child
+			'SiteInfo', // Side tab, 'SiteSetup' child
+			'Languages', // Side tab, 'SiteSetup' child
+			'NavigationMenus', // Side tab, 'SiteSetup' child
+			'SiteTheme', // Side tab, 'SiteAppearance' child
+			'SiteAppearance', // Side tab, 'SiteAppearance' child
+		];	
+
+		$contextDao = Application::getContextDAO();
+		$contextFactory = $contextDao->getAll();
```
[the remark this answers is not in the document]
Comment (defstat) [reply pages/admin/AdminHandler.inc.php:129]: Fixed.

File: pages/admin/AdminHandler.inc.php [hunk 3]
```diff
@@ -118,10 +102,45 @@ function settings($args, $request) {
...
+		$tabs = [
+			'SiteSetup', // Horizontal, parent tab
+			'SiteAppearance', // Horizontal, parent tab
+			'SitePlugins', // Horizontal
+			'SiteConfig', // Side tab, 'SiteSetup' child
+			'SiteInfo', // Side tab, 'SiteSetup' child
+			'Languages', // Side tab, 'SiteSetup' child
+			'NavigationMenus', // Side tab, 'SiteSetup' child
+			'SiteTheme', // Side tab, 'SiteAppearance' child
+			'SiteAppearance', // Side tab, 'SiteAppearance' child
```
[the remark this answers is not in the document]
Comment (defstat) [reply pages/admin/AdminHandler.inc.php:125]: Fixed

File: pages/admin/AdminHandler.inc.php [hunk 4]
```diff
@@ -118,10 +102,45 @@ function settings($args, $request) {
...
 		$templateMgr->assign('settingsData', $settingsData);
+		$templateMgr->assign('componentAvailability', $this->siteSettingsAvailability($request));
 
 		$templateMgr->display('admin/settings.tpl');
 	}
 
+	private function siteSettingsAvailability($request) {
+		$tabsSingleContextAvailability = [
+			'SiteSetup', 
+			'Languages',
```
[the remark this answers is not in the document]
Comment (defstat) [reply pages/admin/AdminHandler.inc.php:113]: Fixed

File: templates/admin/settings.tpl (hunk 1 above)
Comment (defstat) [reply templates/admin/settings.tpl:80]: I regard the availability of the tabs to be a "business rule" and I tend to add the business rules in the handler (keeping the template as clean as possible). 

Also I thought that we could somehow pass this array to the corresponding tabs, thus erasing the multiple `{if}` statements.

> With the component availability approach, it's possible to accidentally get yourself into a case where a parent tab is meant to be hidden but a child tab is meant to be shown.
My initial approach was to actually create a structure holding the data connections that describe the parent-child relationships, and add the code that would fix that. But I thought I would add unnecessary complexity, given that the template it self would not use that structure. Also someone could argue about the hierarchy of things (which availability has the priority, parent's or child's? Effectively the parent's availability priority has been chosen. One drawback of that is that we could end-up with an empty tab - Maybe that's something that needs fixing.

Comment (NateWr) [reply templates/admin/settings.tpl:80]: Ok, let's leave it as-is.

File: pages/admin/AdminHandler.inc.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (NateWr) [reply pages/admin/AdminHandler.inc.php:113]: Have you pushed the changes? They still look PascalCase to me...

[the remark this answers is not in the document]
Comment (defstat) [reply pages/admin/AdminHandler.inc.php]: @NateWr Just now, I was waiting for the last comment response.

**Pull request merged**